### PR TITLE
feat: exchange

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #MF-5735

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/.webpack/config.ts [hunk 1]
```diff
@@ -152,6 +152,11 @@ export async function createConfiguration(_inputFlags: BuildFlags): Promise<webp
                         },
                     },
                 },
+                {
+                    test: /\.svg$/,
+                    include: /node_modules[\\/]@lifi[\\/]wallet-management/, // 只对 node_modules 下的 @lifi/wallet-management 生效
```
Comment (Jack-Works) [inline packages/mask/.webpack/config.ts]: English comment

**Review by Jack-Works — CHANGES_REQUESTED** [review]
please add comments to patches/README.md

File: packages/mask/popups/pages/Swap/index.tsx [hunk 2]
```diff
@@ -85,9 +84,9 @@ export default function SwapPage() {
                         </header>
                         <main className={classes.main}>
                             <EVMWeb3ContextProvider chainId={chainId} networkType={network?.type}>
-                                <AllProviderTradeContext.Provider>
+                                {/* <AllProviderTradeContext.Provider>
                                     <SwapBox />
-                                </AllProviderTradeContext.Provider>
+                                </AllProviderTradeContext.Provider> */}
```
Comment (guanbinrui) [inline packages/mask/popups/pages/Swap/index.tsx]: Remove commented code.

File: packages/plugins/Trader/package.json [hunk 3]
```diff
@@ -16,6 +16,8 @@
     }
   },
   "dependencies": {
+    "@ethersproject/providers": "^5.7.2",
```
Comment (guanbinrui) [inline packages/plugins/Trader/package.json:19]: Can we just compose an object to fit the provider interface from this package? If so we can remove this deps here.

File: packages/web3-hooks/evm/src/index.ts [hunk 4]
```diff
@@ -11,8 +11,6 @@ export * from './useERC721TokenContract.js'
 export * from './useERC721TokenTransferCallback.js'
 export * from './useGasConfig.js'
 export * from './useGasLimit.js'
-export * from './useMulticall.js'
-export * from './useMulticallContract.js'
```
Comment (guanbinrui) [inline packages/web3-hooks/evm/src/index.ts:15]: Please keep these two hook, we may need them in other place.

Comment (guanbinrui) [reply packages/web3-hooks/evm/src/index.ts:15]: We could use wagmi stead.

File: packages/web3-providers/src/Web3/EVM/apis/ContractReadonlyAPI.ts [hunk 5]
```diff
@@ -30,8 +28,6 @@ import ERC1155ABI from '@masknet/web3-contracts/abis/ERC1155.json'
 import CryptoPunksABI from '@masknet/web3-contracts/abis/CryptoPunks.json'
 import WalletABI from '@masknet/web3-contracts/abis/Wallet.json'
 import MulticallABI from '@masknet/web3-contracts/abis/Multicall.json'
-import RouterV2ABI from '@masknet/web3-contracts/abis/RouterV2.json'
-import SwapRouterABI from '@masknet/web3-contracts/abis/SwapRouter.json'
 import WETH_ABI from '@masknet/web3-contracts/abis/WETH.json'
 import PairABI from '@masknet/web3-contracts/abis/Pair.json'
 import QuoterABI from '@masknet/web3-contracts/abis/Quoter.json'
```
Comment (guanbinrui) [inline packages/web3-providers/src/Web3/EVM/apis/ContractReadonlyAPI.ts]: `PairABI` and `QuoterAPI ` are trader related either.

File: packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeInjection.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+export function ExchangeInjection() {
+    const [open, setOpen] = useState(false)
+
+    const handleClose = useCallback(() => setOpen(false), [])
+
+    useEffect(() => {
+        return CrossIsolationMessages.events.swapDialogEvent.on(({ open }) => {
+            setOpen(open)
+        })
+    })
```
Comment (guanbinrui) [inline packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeInjection.tsx]:
```suggestion
    }, [])
```

File: packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+            widgetRef.current?.navigateBack?.()
+        }
+    }, [onClose])
+
+    const getSigner = useCallback(() => {
+        const provider = EVMWeb3.getWeb3Provider({ providerType })
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-expect-error
+        return new Web3Provider(provider, 'any').getSigner()
```
Comment (guanbinrui) [inline packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx:63]: Does `getSigner` return a function?

Comment (guanbinrui) [inline packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx]: Did you mean this?

```suggestion
        return new Web3Provider(provider, 'any').getSigner
```

File: packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    }, [providerType])
+
+    const widgetConfig = useMemo<WidgetConfig>(() => {
+        const rpcs = reduce(
```
Comment (guanbinrui) [inline packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx:59]: Does `lifi` integrated RPCs? If so we could leave `sdkConfig` empty.

File: packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+    const widgetRef = useRef<{
+        navigateToTransaction?: () => void
+        navigateToSettings?: () => void
+        navigateBack?: () => void
+        isOpen(): void
+        toggleDrawer(): void
+        openDrawer(): void
+        closeDrawer(): void
+        isHome: boolean
+    }>(null)
```
Comment (guanbinrui) [inline packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx]: Try to use type from lifi SDK.

File: packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx (hunk 8 above)
Comment (nuanyang233) [reply packages/plugins/Trader/src/SiteAdaptor/trader/ExchangeDialog.tsx:59]: Yes, they have integrated. But I've noticed that sometimes their RPC would crash, causing the balance of the token list to not display. So here I replaced it with our `rpcsUrl`.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**